app/javascript/angular/code/services/_sessions_utils.js

```diff
@@ -97,6 +97,15 @@ export const sessionsUtils = (
     createSessionData(session, data);
     session.loaded = true;
     callback();
+  },
+  refreshMapView: function(sessions) {
+    if (sessions.type === "MobileSessions") {
+      sessions.onSessionsFetchWithCrowdMapLayerUpdate();
+    } else if (sessions.type === "FixedSessions") {
+      sessions.onSessionsFetch();
+    } else {
+      console.warn("Incorrect sessions type");
```

Comment:
when can this happen?

Reply:
Well, this should never happen. But I don't feel comfortable with:
```
    if (sessions.type === "MobileSessions") {
      sessions.onSessionsFetchWithCrowdMapLayerUpdate();
    } else {
      sessions.onSessionsFetch();
    } 
```
when it's clear that the second case should only happen for `(sessions.type === "FixedSessions")`

Reply:
I don't mind checking for the other type, but if there's no other option, don't you think it's misleading cos it suggests there is?

Reply:
But if there is no default branch, that could lead to silent errors if eg the type is misspelled or null. And it's so hard to track what's going on in this frontend, I don't want to be searching for a cause of some unexpected behavior in the future.

How would you write this?